lib/ecto/uuid.ex

```diff
@@ -227,4 +228,5 @@ defmodule Ecto.UUID do
   defp e(13), do: ?d
   defp e(14), do: ?e
   defp e(15), do: ?f
+  defp e(_),  do: throw(:error)
```

Comment:
It is impossible to have any other value, so I am not a big of adding this to appease dialyzer. :)

Reply:
Ok, but dialyzer or not, the catch clause doesn't do anything and it seems to be a remnant and it's there because otherwise the compiler would issue a warning. Shouldn't it be refactored into:
```elixir
@spec encode(raw) :: {:ok, t}
defp encode(...) do
  uuid =
    ...

  {:ok, uuid}
end
```

Reply:
Ah yes, we should be able to remove the try/catch. Salvaging sounds good to me!